core/src/test/scala/zio/config/ArgsListAccumulationTest.scala

```diff
@@ -0,0 +1,41 @@
+package zio.config
+
+import zio.ZIO
+import zio.config.ConfigDescriptor.{ int, list }
+import zio.test.Assertion._
+import zio.test.environment.TestEnvironment
+import zio.test.{ DefaultRunnableSpec, _ }
+
+object ArgsListAccumulationTest extends DefaultRunnableSpec {
```

Comment:
My idea for this test was to test all the applicable input media, like properties, args etc.
`ArgsListAccumulationTest` should be `ListAccumulationTest`

Reply:
Not sure. You were using a different set of combinators + logic to form the args source which is unlike properties, hocon.

Reply:
If the assumption was, every source is a map, then we have a single test case. But its not the case.

Reply:
For example, the above logic for command line args, we are not using Map anymore.
Or to put it this way, every source has their own way of forming the tree.
For example the way we form the tree from hocon isn't the same as we form the tree from a map/sys.env.

Hence we need to test per source.

Ideally, combinators in PropertyTree are proven to be working. But whether the source is making use of the correct combinators lead to testing them

Reply:
PropertyTree.mergeAll was missing from ArgsSource. Now that we added, and we test that explicitly.